DuggaSys/tests/showDuggaservice_test.php

```diff
@@ -0,0 +1,160 @@
+            'opt' => 'UPDATEAU',
+            'hash' => 'hnf3j58s'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'debug',
+            'readonly'
+        )),
+    ),
+    'Create active users' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'UPDATEAU',
+            'hash' => 'tj7dh2nb',
+            'AUtoken' => '999'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'update active users' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'UPDATEAU',
+            'hash' => 'hjk4ert6',
+            'AUtoken' => '9999'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'get data from userAnswer' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
```

Comment:
Where are the inserts and selects needed for this service?
an INSERT INTO course and INSERT INTO listentries is needed for an INSTERT INTO userAnswer.

Every insert needs to be deleted aswell dont forget that!